src/script/api/script_base.cpp

```diff
@@ -10,36 +10,46 @@
 #include "../../stdafx.h"
 #include "script_base.hpp"
 #include "script_error.hpp"
+#include "script_log.hpp"
 
 #include "../../safeguards.h"
 
-/* static */ uint32 ScriptBase::Rand()
+/* static */ SQInteger ScriptBase::Rand()
 {
 	return ScriptObject::GetRandomizer().Next();
 }
 
-/* static */ uint32 ScriptBase::RandItem(int unused_param)
+/* static */ SQInteger ScriptBase::RandItem(SQInteger unused_param)
```

Comment:
unused_param doesn't really need to change.

Reply:
True, but my idea would be to use `SQInteger` everywhere in the API for generic numbers.

Reply:
Then, as a reminder for a later (cleanup) story: remove the casts from squirrel_helper_type.hpp:49-51 as well as 57.

Reply:
Ah you mean squirrel_helper.hpp